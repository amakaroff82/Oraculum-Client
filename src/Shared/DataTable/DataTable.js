import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'recompose';
import ReactTable, { ReactTableDefaults } from 'react-table';
import { withStyles } from 'material-ui/styles';
import { Button } from 'material-ui';
import Loader from '../../Shared/Loader';

/**
This module provides an Apollo-styled version of the ReactTable
component. Accepts all standard props of of ReactTable.
See: https://react-table.js.org/#props

Any passed props will override defaults set in this component.
 */

const styles = theme => ({
  dataTable: {
    width: '100%',
    border: 0,
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    display: 'block',
  },
  tBody: {
    fontSize: theme.typography.body1.fontSize,
    color: theme.palette.text.primary,
  },
  tHeadTr: {
    height: 32,
    '& .rt-th:first-child': {
      paddingLeft: `${theme.spacing.unit * 3}px`,
    },
  },
  header: {
    color: theme.palette.text.primary,
    fontSize: theme.typography.body1.fontSize,
    fontWeight: 600,
    backgroundColor: theme.palette.background.contentFrame,
    whiteSpace: 'pre',
    display: 'flex',
    alignItems: 'center',
  },
  tr: {
    height: 48,
    background: theme.palette.background.paper,
    alignItems: 'center',
    '&:focus': {
      outline: 'none',
    },
    '&:hover, &:focus': {
      background: theme.palette.primary[50],
    },
    '& .rt-td:first-child': {
      paddingLeft: `${theme.spacing.unit * 3}px`,
    },
  },
  cell: {
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    textAlign: 'left',
  },
  pagination: {
    backgroundColor: theme.palette.background.contentFrame,
    fontSize: theme.typography.body1.fontSize,
  },
});

// Provide consistent default filter method
const defaultFilterMethod = (filter, row, column) => {
  const id = filter.pivotId || filter.id;
  return row[id] !== undefined || row[id] !== ''
    ? String(row[id])
        .toLowerCase()
        .includes(filter.value.toLowerCase())
    : true;
};

export const DataTable = ({ classes, intl, ...passThroughProps }) => {
  const PrevComponent = props => {
    return (
      <Button {...props}>
          {"<<<"}
      </Button>
    );
  };

  const NextComponent = props => {
    return (
      <Button {...props}>
          {">>>"}
      </Button>
    );
  };

  // Override ReactTable defaults
  Object.assign(ReactTableDefaults.column, {
    headerClassName: classes.header,
    className: classes.cell,
  });

  // Add props for styling.
  // Props explicity passed to DataTable will override these.
  const reactTableProps = {
    className: classes.dataTable,
    resizable: true,
    showPagination: true,
    defaultPageSize: 10,
    defaultFilterMethod: defaultFilterMethod,
    getTbodyProps: () => ({ className: classes.tBody }),
    getTheadTrProps: () => ({ className: classes.tHeadTr }),
    getTrProps: () => ({ className: classes.tr }),
    getPaginationProps: () => ({ className: classes.pagination }),
    LoadingComponent: ({ loading }) => {
      return loading && <Loader isInner />;
    },
    PreviousComponent: PrevComponent,
    NextComponent: NextComponent,

    pageText: "page",
    ofText: "of",
    rowsText: "rowsText",
    noDataText: "no Results",
    ...passThroughProps,
  };

  return <ReactTable {...reactTableProps} />;
};

DataTable.propTypes = {
  classes: PropTypes.object.isRequired,
};

// Added due to issue describe here:
// https://github.com/adriantoine/enzyme-to-json/issues/19#issuecomment-285781119
DataTable.displayName = 'DataTable';

const enhance = compose(withStyles(styles, { name: 'DataTable' }));

export default enhance(DataTable);
