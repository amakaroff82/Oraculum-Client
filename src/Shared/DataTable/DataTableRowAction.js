import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import IconButton from 'material-ui/IconButton';

const styles = theme => ({
  root: {
    background: 'transparent',
    display: 'inline-block',
  },
  icon: {
    cursor: 'pointer',
    '&, & a': {
      color: theme.palette.text.disabled,
    },
    '.rt-tr:hover &, .rt-tr:hover & a': {
      color: theme.palette.primary[500],
    },
    '.rt-tr &:hover, .rt-tr &:hover a': {
      color: theme.palette.primary[900],
    },
  },
  iconDisabled: {
    '.rt-tr:hover &, .rt-tr:hover & a': {
      color: `${theme.palette.text.disabled}`,
    },
  },
});

export const DataTableRowAction = ({
  childComponent,
  text,
  onClick,
  classes,
  disabled,
}) => {
  if (onClick) {
    return (
      <span className={classes.root}>
        <IconButton
          onClick={onClick}
          classes={{ root: disabled ? classes.iconDisabled : classes.icon }}
          title={text}
          disabled={disabled}
        >
          {childComponent}
        </IconButton>
      </span>
    );
  } else {
    return (
      <span className={disabled ? classes.iconDisabled : classes.icon}>
        {childComponent}
      </span>
    );
  }
};

DataTableRowAction.propTypes = {
  childComponent: PropTypes.object,
  text: PropTypes.string,
  onClick: PropTypes.any,
  disabled: PropTypes.bool,
};

export default withStyles(styles, { name: 'DataTableRowAction' })(
  DataTableRowAction
);
