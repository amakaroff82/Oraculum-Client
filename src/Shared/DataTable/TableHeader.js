import React from 'react';
import { Grid, Toolbar } from 'material-ui';
import Typography from 'material-ui/Typography';
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
  root: {
    height: theme.spacing.unit * 12,
  },
});

export const TableHeader = ({
  classes,
  headerLabel,
  leftButtons,
  rightButtons,
}) => {
  return (
    <Grid container alignItems="center" wrap="nowrap" className={classes.root}>
      <Grid item xs={6}>
        <Toolbar>
          <Typography type="title">{headerLabel}</Typography>
          <Toolbar>{leftButtons}</Toolbar>
        </Toolbar>
      </Grid>
      <Grid item xs={6}>
        <Grid container justify="flex-end">
          <Grid item>
            <Toolbar>{rightButtons}</Toolbar>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default withStyles(styles, { name: 'TableHeader' })(TableHeader);
