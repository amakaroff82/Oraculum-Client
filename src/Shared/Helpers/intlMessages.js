export const intlMessages = {
  orchestration: { id: 'Orchestration', defaultMessage: 'Orchestration' },
  products: { id: 'Products', defaultMessage: 'Products' },
  status: { id: 'Status', defaultMessage: 'Status' },
  dateLastModified: {
    id: 'DateLastModified',
    defaultMessage: 'Date last modified',
  },
  title: { id: 'Title', defaultMessage: 'Titre' },
  topic: { id: 'Topic', defaultMessage: 'Topic' },
  createdBy: { id: 'CreatedBy', defaultMessage: 'Created by' },
  version: { id: 'Version', defaultMessage: 'Version' },
};

export default intlMessages;
