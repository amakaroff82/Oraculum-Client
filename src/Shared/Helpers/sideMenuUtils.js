import React from 'react';
import { ListItem, ListItemText } from 'material-ui/List';

export const renderListItems = (listItemData, listAction, identifierKey) => {
  if (listItemData && listAction && identifierKey) {
    return listItemData.map((element, index) => {
      return (
        <ListItem
          key={element[identifierKey]}
          button
          onClick={() => listAction(element)}
        >
          <ListItemText color="primary" primary={element[identifierKey]} />
        </ListItem>
      );
    });
  }
};
