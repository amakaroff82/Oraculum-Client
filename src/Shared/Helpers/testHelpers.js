// Create mock response for testing
export const makeMockResponse = (
  response,
  status = 200,
  statusText = 'OK',
  headers = { 'Content-Type': 'application/json' }
) => {
  const init = {
    status,
    statusText,
    headers,
  };
  return new window.Response(JSON.stringify(response), init);
};

export const jsonParseError = () => {
  try {
    JSON.parse('+data+');
  } catch (err) {
    return err;
  }
};

export const makeFakeError = (
  response = null,
  status = 400,
  statusText = 'Bad Request'
) => {
  const error = new Error('Error');
  error.response = makeMockResponse(response, status, statusText);
  return error;
};
