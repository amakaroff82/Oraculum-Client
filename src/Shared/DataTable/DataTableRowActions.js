import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';

import DataTableRowAction from './DataTableRowAction';

const styles = {
  root: {
    display: 'block',
  },
};

export const DataTableRowActions = ({ actions, classes }) => {
  return (
    <div className={classes.root}>
      {actions &&
        actions.map((action, index) => {
          return <DataTableRowAction key={index} {...action} />;
        })}
    </div>
  );
};

DataTableRowActions.propTypes = {
  actions: PropTypes.array.isRequired,
};

export default withStyles(styles, { name: 'DataTableRowActions' })(
  DataTableRowActions
);
