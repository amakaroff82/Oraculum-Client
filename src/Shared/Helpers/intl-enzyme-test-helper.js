/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */

import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { intlMessages } from './intlMessages';

