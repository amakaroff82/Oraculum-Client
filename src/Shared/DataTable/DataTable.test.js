import * as React from 'react';
import { shallowWithIntl } from '../../Shared/Helpers/intl-enzyme-test-helper';
import toJson from 'enzyme-to-json';
import { mockJourneys, mockColumns } from '../../Journeys/Helpers/mocks';
import { DataTable } from './DataTable';

const defaultProps = {
  classes: {},
  data: mockJourneys,
  columns: mockColumns,
};

const setup = setupProps => {
  const props = { ...defaultProps, ...setupProps };
  return shallowWithIntl(<DataTable {...props} />);
};

describe('<DataTable />', () => {
  it('renders snapshot correctly with required props', () => {
    const tree = setup();
    expect(toJson(tree)).toMatchSnapshot();
  });
});
