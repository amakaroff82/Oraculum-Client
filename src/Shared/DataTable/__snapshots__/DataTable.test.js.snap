// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTable /> renders snapshot correctly with required props 1`] = `
<ReactTable
  AggregatedComponent={[Function]}
  ExpanderComponent={[Function]}
  FilterComponent={[Function]}
  LoadingComponent={[Function]}
  NextComponent={[Function]}
  NoDataComponent={[Function]}
  PadRowComponent={[Function]}
  PaginationComponent={[Function]}
  PivotValueComponent={[Function]}
  PreviousComponent={[Function]}
  ResizerComponent={[Function]}
  TableComponent={[Function]}
  TbodyComponent={[Function]}
  TdComponent={[Function]}
  TfootComponent={[Function]}
  ThComponent={[Function]}
  TheadComponent={[Function]}
  TrComponent={[Function]}
  TrGroupComponent={[Function]}
  aggregatedKey="_aggregated"
  className=""
  collapseOnDataChange={true}
  collapseOnPageChange={true}
  collapseOnSortingChange={true}
  column={
    Object {
      "Aggregated": undefined,
      "Cell": undefined,
      "Expander": undefined,
      "Filter": undefined,
      "Footer": undefined,
      "Header": undefined,
      "Pivot": undefined,
      "PivotValue": undefined,
      "aggregate": undefined,
      "className": undefined,
      "filterAll": false,
      "filterMethod": undefined,
      "filterable": undefined,
      "footerClassName": "",
      "footerStyle": Object {},
      "getFooterProps": [Function],
      "getHeaderProps": [Function],
      "getProps": [Function],
      "headerClassName": undefined,
      "headerStyle": Object {},
      "minWidth": 100,
      "resizable": undefined,
      "show": true,
      "sortMethod": undefined,
      "sortable": undefined,
      "style": Object {},
    }
  }
  columns={
    Array [
      Object {
        "Header": "Name",
        "accessor": "name",
      },
      Object {
        "Header": "Status",
        "accessor": "status",
      },
      Object {
        "Header": "Topic",
        "accessor": "topic",
      },
      Object {
        "Header": "Products",
        "accessor": [Function],
        "id": "products",
      },
      Object {
        "Header": "Date Last Modified",
        "accessor": "dateLastModified",
      },
      Object {
        "Header": "Orchestration",
        "accessor": "crmOrchestration",
      },
    ]
  }
  data={
    Array [
      Object {
        "crmOrchestration": "Triggerable",
        "dateLastModified": "2017-05-10T15:16:25.087",
        "id": "1",
        "name": "Prelaunch Teaser",
        "owner": "swift_nick@yahoo.com",
        "products": Array [
          "Atrino Health",
        ],
        "status": "Running",
        "topic": "General Communications",
      },
      Object {
        "crmOrchestration": "Not Shared",
        "dateLastModified": "2017-05-16T16:36:38.19",
        "id": "2",
        "name": "Physician Event",
        "owner": "jedidiah.fahey@george.ca",
        "products": Array [
          "Atrino Health",
        ],
        "status": "Running",
        "topic": "Event Nurture Info",
      },
      Object {
        "crmOrchestration": "Viewable",
        "dateLastModified": "2017-05-16T16:56:52.977",
        "id": "3",
        "name": "Q2 Campaigns",
        "owner": "daphne_leuschke@hotmail.com",
        "products": Array [
          "Atrino Health",
        ],
        "status": "Draft",
        "topic": "Web Contacts",
      },
      Object {
        "crmOrchestration": "Not Shared",
        "dateLastModified": "2017-05-18T16:47:49.56",
        "id": "4",
        "name": "Austin Symposium",
        "owner": "ida.walker@brown.biz",
        "products": Array [
          "Atrino Health",
        ],
        "status": "Running",
        "topic": "Event Nurture Info",
      },
      Object {
        "crmOrchestration": "Triggerable",
        "dateLastModified": "2017-05-22T15:04:13.57",
        "id": "5",
        "name": "Objection blunting",
        "owner": "bria.jewess@yahoo.com",
        "products": Array [
          "Atrino Health",
        ],
        "status": "Running",
        "topic": "Sales Communications",
      },
      Object {
        "crmOrchestration": "Not Shared",
        "dateLastModified": "2017-05-22T17:43:25.24",
        "id": "6",
        "name": "Objection blunting",
        "owner": "bartell_alicia@langworth.ca",
        "products": Array [
          "Atrino Health",
        ],
        "status": "Draft",
        "topic": "General Communications",
      },
    ]
  }
  defaultExpanded={Object {}}
  defaultFilterMethod={[Function]}
  defaultFiltered={Array []}
  defaultPageSize={10}
  defaultResized={Array []}
  defaultSortDesc={false}
  defaultSortMethod={[Function]}
  defaultSorted={Array []}
  expanderDefaults={
    Object {
      "filterable": false,
      "resizable": false,
      "sortable": false,
      "width": 35,
    }
  }
  filterable={false}
  freezeWhenExpanded={false}
  getLoadingProps={[Function]}
  getNoDataProps={[Function]}
  getPaginationProps={[Function]}
  getProps={[Function]}
  getResizerProps={[Function]}
  getTableProps={[Function]}
  getTbodyProps={[Function]}
  getTdProps={[Function]}
  getTfootProps={[Function]}
  getTfootTdProps={[Function]}
  getTfootTrProps={[Function]}
  getTheadFilterProps={[Function]}
  getTheadFilterThProps={[Function]}
  getTheadFilterTrProps={[Function]}
  getTheadGroupProps={[Function]}
  getTheadGroupThProps={[Function]}
  getTheadGroupTrProps={[Function]}
  getTheadProps={[Function]}
  getTheadThProps={[Function]}
  getTheadTrProps={[Function]}
  getTrGroupProps={[Function]}
  getTrProps={[Function]}
  groupedByPivotKey="_groupedByPivot"
  indexKey="_index"
  loading={false}
  loadingText="Loading..."
  nestingLevelKey="_nestingLevel"
  nextText="Next"
  noDataText={
    <FormattedMessage
      id="noResults"
      values={Object {}}
    />
  }
  ofText={
    <FormattedMessage
      id="of"
      values={Object {}}
    />
  }
  onFetchData={[Function]}
  originalKey="_original"
  pageSizeOptions={
    Array [
      5,
      10,
      20,
      25,
      50,
      100,
    ]
  }
  pageText={
    <FormattedMessage
      id="page"
      values={Object {}}
    />
  }
  pivotDefaults={Object {}}
  pivotIDKey="_pivotID"
  pivotValKey="_pivotVal"
  previousText="Previous"
  resizable={true}
  rowsText="rowsText"
  showPageJump={true}
  showPageSizeOptions={true}
  showPagination={true}
  showPaginationBottom={true}
  showPaginationTop={false}
  sortable={true}
  style={Object {}}
  subRowsKey="_subRows"
/>
`;
